cirq/devices/grid_qubit.py

```diff
@@ -82,6 +64,197 @@ def neighbors(self,
+
+        Returns:
+            A list of GridQubits filling in a square grid
+        """
+        return GridQid.rect(diameter, diameter, top=top, left=left, dimension=dimension)
+
+    @staticmethod
+    def rect(rows: int, cols: int, top: int = 0, left: int = 0,  *, dimension: int) -> List['GridQid']:
+        """Returns a rectangle of GridQubits.
+
+        Args:
+            rows: Number of rows in the rectangle
+            cols: Number of columns in the rectangle
+            top: Row number of the topmost row
+            left: Column number of the leftmost row
+            dimension: The dimension of the qid, e.g. the number of quantum
+                levels.
+
+        Returns:
+            A list of GridQubits filling in a rectangular grid
+        """
+        return [
+            GridQid(row, col, dimension=dimension)
+            for row in range(top, top + rows)
+            for col in range(left, left + cols)
+        ]
+
+    @staticmethod
+    def from_diagram(diagram: str, dimension: int) -> List['GridQid']:
+        """Parse ASCII art device layout into info about qids and
+        connectivity. As an example, the below diagram will create a list of
+        GridQid in a pyramid structure.
+        ---A---
+        --AAA--
+        -AAAAA-
+        AAAAAAA
+
+        You can use any character other than a hyphen to mark a qid. As an
+        example, the qids for the Bristlecone device could be represented by
+        the below diagram. This produces a diamond-shaped grid of qids, and
+        qids with the same letter correspond to the same readout line.
+
+        .....AB.....
+        ....ABCD....
+        ...ABCDEF...
+        ..ABCDEFGH..
+        .ABCDEFGHIJ.
+        ABCDEFGHIJKL
+        .CDEFGHIJKL.
+        ..EFGHIJKL..
+        ...GHIJKL...
+        ....IJKL....
+        .....KL.....
+
+        Args:
+            diagram: String representing the qid layout. Each line represents
+                a row. Alphanumeric characters are assigned as qid.
+                Dots ('.'), dashes ('-'), and spaces (' ') are treated as
+                empty locations in the grid. If diagram has characters other
+                than alphanumerics, spacers, and newlines ('\n'), an error will
+                be thrown. The top-left corner of the diagram will be have
+                coordinate (0,0).
+
+        Returns:
+            A list of GridQid corresponding to qids in the provided diagram
+
+        Raises:
+            ValueError: If the input string contains an invalid character.
+        """
+        coords = _ascii_diagram_to_coords(diagram)
+        return [GridQid(*c, dimension=dimension) for c in coords]
+
+    def __repr__(self):
+        return 'cirq.GridQid({}, dimension={})'.format(self.x, self.dimension)
+
+    def __str__(self):
+        return '({}, {}) (d={})'.format(self.row, self.col, self.dimension)
+
+    def _json_dict_(self):
+        return protocols.obj_to_dict_helper(self, ['x', 'dimension'])
```

Comment:
row, col, dimension

You'll also need to add an entry into the json serialization code. (The tests will enforce that this happens.)